editor/sidebar/post-taxonomies/index.js

```diff
@@ -35,9 +35,11 @@ class PostTaxonomies extends Component {
 
 	componentDidMount() {
 		this.fetchTaxonomies = new wp.api.collections.Taxonomies()
-			.fetch()
+			.fetch( { data: { context: 'edit' } } )
 			.done( ( taxonomies ) => {
-				this.setState( { taxonomies: Object.values( taxonomies ) } );
+				this.setState(
+					{ taxonomies: Object.values( taxonomies ).filter( taxonomy => taxonomy.public ) }
```

Comment:
Minor: Lodash's `filter` handles this beautifully (coercing object to arrange, filtering on key):

```js
this.setState( {
	taxonomies: filter( taxonomies, { public: true } ),
} );
```

Reply:
Was thinking about using Lodash, this looks perfect!